### ISSUE TITLE
Issue with es6 functionalities

### ISSUE THREAD

**Issue by chiel** [talmobi/wrollup#1]
Just trying this module now while waiting for rollup-plugin-watch to get fixed, but ran into this issue:

```
/Users/chiel/dev/cms/node_modules/wrollup/index.js:254
    for (let i = 0; i < bundle.modules.length; i++) {
         ^^^

SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/Users/chiel/dev/cms/node_modules/wrollup/cli.js:2:1)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
```

Is this perhaps because I am not on node v6? On v4.4.7 at the moment.

**Comment by talmobi** [comment]
Hi, indeed!

I added 'use strict' and tested with node v 4.4.7 and it works now!

Does it work for you? Try the latest version 0.1.5

``` js
npm install wrollup@latest
```

Thanks for the report!

**Comment by chiel** [comment]
I've installed install node 6.7.0 now and tested it there - it seems to work fine - thanks!

I am running into another issue now though, not sure if that's something that your package would fix, but it's the maximum call stack size exceeded thing again:

```
Error: Could not load react (imported by /Users/chiel/dev/cms/src/app/components/App.jsx): ENOENT: no such file or directory, open 'react'
    at /Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:8775:10
RangeError: Maximum call stack size exceeded
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:153:20)
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:165:18)
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:165:18)
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:165:18)
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:165:18)
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:165:18)
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:165:18)
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:165:18)
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:165:18)
    at deepClone (/Users/chiel/dev/cms/node_modules/rollup/dist/rollup.js:165:18)
```

**Comment by talmobi** [comment]
hmm,

does

``` js
rollup -c rollup.config.js
```

by itself work?

Maybe you need to use the rollup-plugin-node-resolve plugin?

This is what my sample rollup.config looks like:

``` js
import buble from 'rollup-plugin-buble'
import nodeResolve from 'rollup-plugin-node-resolve'
import commonjs from 'rollup-plugin-commonjs'

export default {
  entry: 'src/index.js',
  dest: 'dist/bundle.js',
  format: 'iife',
  plugins: [
    buble(),
    nodeResolve(),
    commonjs({
      include: 'node_modules/**'
    })
  ]
}

```

You said that it works? when does the error occur?

Thanks for the input!

**Comment by chiel** [comment]
Yeah rollup (and wrollup) works fine on initial bundling, but throws an error on subsequent builds when there are external dependencies.

**Comment by talmobi** [comment]
Hmmm!! I'd like to reproduce this error if possible, could you paste your rollup.config.js and App.jsx for me?

Thanks!

**Comment by chiel** [comment]
Of course.

### `rollup.server.js`

``` js
import babel from 'rollup-plugin-babel';

export default {
    entry: 'src/server/index.jsx',
    dest: 'dist/server/index.js',
    format: 'cjs',
    plugins: [
        babel(),
    ],
}
```

### `app/components/App.jsx`

``` jsx
import React from 'react';

export default function App() {
    return (
        <p>Hello, world</p>
    );
}
```

### `server/index.jsx`

``` jsx
import App from '../app/components/App.jsx';
import express from 'express';
import React from 'react';
import { renderToString } from 'react-dom/server';
import renderFullPage from './utils/renderFullPage';

const app = express();

app.get('/', (req, res) => {
    res.send(renderToString(<App />));
});

const port = process.env.PORT || 11002;
app.listen(port, () => console.info(`Listening on port ${port}`));
```

### dependencies

```
npm i -S express react react-dom serve-static
```

When saving something in `server/index.jsx`, everything seems just fine, when saving in `app/components/App.jsx`, everything seems to go to hell. But yeah, this happens even outside of `wrollup`, so it's not really something wrong there.

**Comment by talmobi** [comment]
It's an issue of how rollup uses caching of subsequent bundles. I'm not entirely 100% sure the way you're doing is the "intended" way, however, I'm currently implementing a work around that works (basically an option to disable bundle caching).

**Comment by chiel** [comment]
Yeah to be honest, I'm not sure that I should be using rollup for my server code, but it seemed like a good idea. I'm gonna go back to just having babel take care of the server code, and then having rollup handle client-side only. The downside of this approach is that I will need to configure babel differently for server and client - something I was hoping to avoid.

Thanks anyway for your insights!

**Comment by talmobi** [comment]
Ok, could you try version 0.1.6 with the --nocache flag to disable caching.

``` js
npm install -g wrollup@latest
wrollup -c rollup.config.js --nocache
```

**Comment by talmobi** [comment]
@chielkunkels

Did the `--nocache` work around fix the issue?

Also, which rollup version are you using? Another user (see https://github.com/talmobi/wrollup/issues/2) had trouble with the latest rollup version 0.36.1 that were, at least for now, solved by using rollup version 0.34.10 instead.
